pkg/observable/channel/observable_test.go

```diff
@@ -0,0 +1,366 @@
+package channel_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"pocket/internal/testchannel"
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/channel"
+)
+
+const (
+	productionDelay        = 2 * time.Millisecond
+	notifyTimeout          = productionDelay * 5
+	cancelUnsubscribeDelay = notifyTimeout * 2
+)
+
+func TestChannelObservable_NotifyObservers(t *testing.T) {
+	type test struct {
+		name            string
+		producer        chan *int
+		inputs          []int
+		expectedOutputs []int
+		setupFn         func(t test)
+	}
+
+	inputs := []int{123, 456, 789}
+	// NB: see INCOMPLETE comment below
+	// fullBlockingProducer := make(chan *int)
+	// fullBufferedProducer := make(chan *int, 1)
+
+	tests := []test{
+		{
+			name:            "nil producer",
+			producer:        nil,
+			inputs:          inputs,
+			expectedOutputs: inputs,
+		},
+		{
+			name:            "empty non-buffered producer",
+			producer:        make(chan *int),
+			inputs:          inputs,
+			expectedOutputs: inputs,
+		},
+		{
+			name:            "empty buffered len 1 producer",
+			producer:        make(chan *int, 1),
+			inputs:          inputs,
+			expectedOutputs: inputs,
+		},
+		// INCOMPLETE: producer channels which are full are proving harder to test
+		// robustly (no flakiness); perhaps it has to do with the lack of some
+		// kind of guarantee about the receiver order on the consumer side.
+		//
+		// The following scenarios should generally pass but are flaky:
```

Comment:
Can you add more details (screenshot is fine) on what the error is?

Reply:
![image](https://github.com/pokt-network/poktroll/assets/600733/8408c6d1-475b-42eb-920b-0e10e9899076)

Reply:
Created https://github.com/pokt-network/poktroll/issues/81 and added `TODO_INCOMPLETE(#81)` in the comment here.